lib/internal/errors.js

```diff
@@ -1809,6 +1809,9 @@ E('ERR_UNSUPPORTED_ESM_URL_SCHEME', (url, supported) => {
   msg += `. Received protocol '${url.protocol}'`;
   return msg;
 }, Error);
+E('ERR_UNSUPPORTED_RESOLVE_REQUEST',
+  'Failed to resolve module specifier "%s" from "%s": Invalid relative url or base scheme is not hierarchical.',
```

Comment:
It's not just with `data:` URLs, however. It's really any non-special URL whose hierarchy semantics cannot be determined. For instance, suppose we supported a hypothetical `fake:module` URL in the future, this error would also apply in that case since `new URL('something', 'fake:module')` would end up as an invalid URL error.

One way we could improve the error message is to say something like:

```
Failed to resolve module specifier "...": import.meta.resolve is not supported with ${baseUrl}
```

Where `${baseUrl}` is the full url of the referring module.

Reply:
But `import.meta.resolve` _is_ supported with data: URLs, just it won't accept relative URLs. I inspired myself for this error message from Chromium's error message, which might not be perfect, but is probably at least technically correct